services/tracking-service/reactive-vertx/src/main/java/com/amazon/verticles/RedisVerticle.java

```diff
@@ -22,96 +22,107 @@
-            if (!res.succeeded()) {
-                LOGGER.info(res.cause());
-            }
-        }));
+    void registerToEventBusForPurging(final EventBus eb) {
+        eb
+                .consumer(Constants.REDIS_PURGE_EVENTBUS_ADDRESS)
+                .handler(msg ->
+                        redis
+                                .send(cmd(FLUSHALL))
+                                .onFailure(err -> LOGGER.info(err.getMessage())));
     }
 
-    void registerToEventBusForCacheVerticle(final EventBus eb, final RedisClient redis) {
-        eb.consumer(Constants.REDIS_EVENTBUS_ADDRESS, message -> {
-            // Getting data from Redis and storing it in cache verticle
-
-            TrackingMessage trackingMessage = Json.decodeValue(((JsonObject)message.body()).encode(), TrackingMessage.class);
-            LOGGER.info(RedisVerticle.class.getSimpleName() + ": I have received a message: " + message.body());
-
-            LOGGER.info("Looking for programId " + trackingMessage.getProgramId() + " in Redis");
-
-            redis.hgetall(trackingMessage.getProgramId(), res -> {
-                if (res.succeeded()) {
-                    JsonObject result = res.result();
-                    if (null == result || result.isEmpty()) {
-                        LOGGER.info("No object found");
-                        message.reply(new JsonObject());
-                    } else {
-                        String strRes = Json.encode(result);
-                        TrackingMessage msg = Json.decodeValue(strRes, TrackingMessage.class);
-                        msg.setMessageId(trackingMessage.getMessageId());
-
-                        JsonObject msgResult = JsonObject.mapFrom(msg);
-
-                        LOGGER.info("Result: " + msgResult);
-                        message.reply(msgResult);
-                    }
-                } else {
-                    LOGGER.info("No object found: " + res.cause());
-                    message.reply(new JsonObject());
-                }
-            });
-
-        });
+    void registerToEventBusForCacheVerticle(final EventBus eb) {
+        eb
+                .<JsonObject>consumer(Constants.REDIS_EVENTBUS_ADDRESS)
+                .handler(message -> {
+                    // Getting data from Redis and storing it in cache verticle
+
+                    TrackingMessage trackingMessage = Json.decodeValue(message.body().encode(), TrackingMessage.class);
+                    LOGGER.info(RedisVerticle.class.getSimpleName() + ": I have received a message: " + message.body());
+
+                    LOGGER.info("Looking for programId " + trackingMessage.getProgramId() + " in Redis");
+
+                    redis
+                            .send(cmd(HGETALL).arg(trackingMessage.getProgramId()))
+                            .onSuccess(result -> {
+                                if (null == result || result.size() == 0) {
```

Comment:
Looking at this, would it make sense for the redis API to always return a non null `Response` which would have either a type `NULL` or a getter `null()` that returns true only for this case? This response would be a singleton so in terms of GC it would not behave much differently.

/cc @vietj @smoell

Reply:
I think you can consider this option indeed if both have the same semantic

Reply:
Thank would be helpful, yes